bittensor/subtensor.py

```diff
@@ -1418,12 +1432,18 @@ def _do_serve_axon(
         enhancing the decentralized computation capabilities of Bittensor.
         """
 
+        if call_params["certificate"] is None:
+            del call_params["certificate"]
+            call_function = "serve_axon"
+        else:
+            call_function = "serve_axon_tls"
+
```

Comment:
I guess it was needed for backwards compatibility. A new method was added to substrate, instead of modifying the existing one.

Reply:
it's sad that the placeholders they left in case of stuff like this can't be used https://github.com/backend-developers-ltd/subtensor-fork/blob/development/pallets/subtensor/src/serving.rs#L30

Reply:
We need to validate that the provided certificate length is correct before calling substrate.